src/storage/src/source/postgres.rs

```diff
@@ -298,7 +298,7 @@ pub enum DefiniteError {
 impl From<DefiniteError> for SourceReaderError {
     fn from(err: DefiniteError) -> Self {
         SourceReaderError {
-            inner: SourceErrorDetails::Other(err.to_string()),
+            inner: SourceErrorDetails::Permanent(err.to_string()),
```

Comment:
This in particular feels high-risk... there are lots of definite postgres errors that are not permanent!

Reply:
@petrosagg @sploiselle my understanding is that all of these are permanent?

Reply:
awesome!

Reply:
will update the docs to clarify this!

Reply:
> where the error can be retracted in the future

What is the mechanism by which PG can emit retractions for an error like this?

Reply:
Here we map the raw collection (which might contain diffs of any value, including negatives) with two fallible functions https://github.com/MaterializeInc/materialize/blob/main/src/storage/src/source/postgres/replication.rs#L338-L341

So if the raw collection contains a `-1` diff with invalid utf data then we will correctly retract it from the output too

Reply:
Got it––I had mistakenly assumed this was an error with the data from the replication stream––i.e. it was corrupted. My misunderstanding here.